### ISSUE TITLE
Abort configuration parsing after encountering X errors

### ISSUE THREAD

**Issue by TimWolla** [haproxy/haproxy#625]
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 2.2-dev7-49fecd-22 2020/05/07 - https://haproxy.org/
Status: development branch - not safe for use in production.
Known bugs: https://github.com/haproxy/haproxy/issues?q=is:issue+is:open
Running on: Linux 4.4.0-178-generic #208-Ubuntu SMP Sun Apr 5 23:45:10 UTC 2020 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-old-style-declaration -Wno-ignored-qualifiers -Wno-clobbered -Wno-missing-field-initializers -Wtype-limits
  OPTIONS = 

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT -PCRE2 -PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO -OPENSSL -LUA +FUTEX +ACCEPT4 -ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL -SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=4).
Built with network namespace support.
Built with gcc compiler version 5.4.0 20160609
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built without PCRE or PCRE2 support (using libc's regex instead)
Encrypted password support via crypt(3): yes
Built without compression support (neither USE_ZLIB nor USE_SLZ are set).
Compression algorithms supported : identity("identity")

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2
            fcgi : mode=HTTP       side=BE        mux=FCGI
       <default> : mode=HTTP       side=FE|BE     mux=H1
       <default> : mode=TCP        side=FE|BE     mux=PASS

Available services : none

Available filters :
	[SPOE] spoe
	[CACHE] cache
	[FCGI] fcgi-app
	[TRACE] trace
	[COMP] compression
```

## What should haproxy do differently? Which functionality do you think we should add?

This issue is in response to my patch e6291956e7ad622dd2e4ac864caf05c93db417d3 that fixes a degenerate case in the configuration parser, allowing to generate gigabytes of error output, due to a large number of errors. This is a DoS opportunity with a large autogenerated configuration which might have the same typo within all configured backends.

HAProxy usually carries on parsing as much as possible, reporting all errors at the same type to relieve the user from needing to re-check the configuration after fixing a single error only to see that another error pops up.
However it is not really useful to report more than a dozen errors at once. Human written configurations are expected to be created iteratively, each change making a small change with a limited chance for error, probably not hitting that limit.
Machine generated configuration can easily contain repeated errors if not careful, causing the mentioned DoS issue when attempting to parse the config.

Some hard limit error should be added after which the configuration parsing is aborted, even if there might be more errors.

## What are you trying to do?

see above

**Comment by TimWolla** [comment]
@wtarreau I believe this one is resolved with 32234e751320b60a3879f274d4a4753d7570e757, yes?

**Comment by wtarreau** [comment]
I'd say yes even if you found some limitations to this design.
